internal/parse/parts.go

```diff
@@ -0,0 +1,68 @@
+package parse
+
+// A queryPart represents a section of a parsed SQL statement, which forms
+// a complete query when processed together with its surrounding parts, in
+// their correct order.
+type queryPart interface {
+	// String returns the part's representation for debugging purposes.
+	String() string
+
+	// ToSQL returns the SQL representation of the part.
+	ToSQL() string
+}
+
+// FullName represents a table column or a Go type identifier.
+type FullName struct {
+	Prefix, Name string
+}
+
+// InputPart represents a named parameter that will be send to the database
+// while performing the query.
+type InputPart struct {
+	FullName
+}
+
+// String returns a textual representation of the InputPart meant for debugging
```

Comment:
We don't need these boilerplate docstrings on every one of these methods, as this will be pretty much a copy & paste from the queryPart interface.  As a guideline, we should only document something when the text is actually improving the understanding of the reader about the code at hand. Otherwise, instead of helping we're rather hindering the understanding of the code by forcing the reading of something obvious.

Reply:
Removed comments from methods already in the interface.